lib/chewy/fields/root.rb

```diff
@@ -30,7 +30,7 @@ def mappings_hash
         end
 
         mappings[name][:_parent] = parent.is_a?(Hash) ? parent : {type: parent} if parent
-        mappings
+        mappings[name]
```

Comment:
It is the breaking change. 
If you want to support both 6 and 7 Elastic versions you could pass special request params.

 https://www.elastic.co/guide/en/elasticsearch/reference/current/breaking-changes-7.0.html#hits-total-now-object-search-response
 https://www.elastic.co/blog/moving-from-types-to-typeless-apis-in-elasticsearch-7-0
 https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html#_schedule_for_removal_of_mapping_types

https://github.com/toptal/chewy/issues/673#issuecomment-631432562

Reply:
How does the response look like for pre-7 and for 7+ ? based on that, you could determine if you need to return either `mapping` or `mapping[name]` 

or am I oversimplifying the problem and I'm not taking something in account?